inc/azure_c_shared_utility/platform.h

```diff
@@ -16,6 +16,9 @@ extern "C" {
     MOCKABLE_FUNCTION(, void, platform_deinit);
     MOCKABLE_FUNCTION(, const IO_INTERFACE_DESCRIPTION*, platform_get_default_tlsio);
     MOCKABLE_FUNCTION(, STRING_HANDLE, platform_get_platform_info);
+#ifdef WIN32
```

Comment:
Please remove checks for WIN32 preprocessor within the headers. We like to keep things consistent in headers and have each platform specific header provide the appopriate value for the platform_info.